### PR TITLE
ASL-623 asio update for Orbis (Sony PS4)

### PR DESCRIPTION
**Pull request by roman380**

### CODE REVIEW

File: asio/include/asio/detail/impl/socket_ops.ipp [hunk 1]
```diff
@@ -2484,6 +2533,64 @@ inline hostent* gethostbyaddr(const char* addr, int length, int af,
     return 0;
   *result = *retval;
   return retval;
+#elif defined(__ORBIS__) || defined(__PROSPERO__)
+
+  // TODO: Move this outra here
```
Comment (roman380) [inline asio/include/asio/detail/impl/socket_ops.ipp]: Normalize this

File: asio/include/asio/detail/impl/socket_ops.ipp [hunk 2]
```diff
@@ -2539,6 +2646,53 @@ inline hostent* gethostbyname(const char* name, int af, struct hostent* result,
...
```
Comment (roman380) [inline asio/include/asio/detail/impl/socket_ops.ipp]: Normalize this

File: asio/include/asio/detail/posix_signal_blocker.hpp [hunk 3]
```diff
@@ -37,42 +43,42 @@ class posix_signal_blocker
   posix_signal_blocker()
     : blocked_(false)
   {
-    sigset_t new_mask;
-    sigfillset(&new_mask);
-    blocked_ = (pthread_sigmask(SIG_BLOCK, &new_mask, &old_mask_) == 0);
+    //sigset_t new_mask; // TODO: Figure out how to re-enable it, maybe at least asserts?
```
Comment (roman380) [inline asio/include/asio/detail/posix_signal_blocker.hpp]: Double-check if this is needed or can actually be commented out?

**Review by lultimouomo — COMMENTED** [review]
@roman380 I think we could merge this as-is, but since we use so little of asio, and we were including it for no reason except my laziness, I have created https://github.com/castlabs/common_libs_cross-platform-helpers/pull/202 . I believe once we merge that PR most of the changes in this PR will not be needed, and we might get away with a much smaller diff against upstream. What do you think?

**Comment by roman380** [comment]
Reverted a few files which are no longer needed because of https://github.com/castlabs/common_libs_cross-platform-helpers/pull/202, also grouped commits for reasonable commit history; builds with both PS4 & PS5 SDKs

**Review by lultimouomo — APPROVED** [review]

**Pull request merged**